x-pack/plugins/observability_solution/slo/server/services/create_slo.ts

```diff
@@ -47,62 +48,41 @@ export class CreateSLO {
+    rollbackOperations.push(() => this.repository.deleteById(slo.id, true));
 
     const rollupTransformId = getSLOTransformId(slo.id, slo.revision);
     const summaryTransformId = getSLOSummaryTransformId(slo.id, slo.revision);
     try {
-      await retryTransientEsErrors(
-        () =>
-          this.scopedClusterClient.asSecondaryAuthUser.ingest.putPipeline(
-            getSLOPipelineTemplate(slo)
-          ),
-        { logger: this.logger }
-      );
-      rollbackOperations.push(() =>
-        this.scopedClusterClient.asSecondaryAuthUser.ingest.deletePipeline(
-          { id: getSLOPipelineId(slo.id, slo.revision) },
-          { ignore: [404] }
-        )
-      );
+      const sloPipelinePromise = this.createPipeline(getSLOPipelineTemplate(slo));
+      rollbackOperations.push(() => this.deletePipeline(getSLOPipelineId(slo.id, slo.revision)));
 
-      await this.transformManager.install(slo);
+      const rollupTransformPromise = this.transformManager.install(slo, true);
       rollbackOperations.push(() => this.transformManager.uninstall(rollupTransformId));
-
-      await this.transformManager.start(rollupTransformId);
       rollbackOperations.push(() => this.transformManager.stop(rollupTransformId));
 
-      await retryTransientEsErrors(
-        () =>
-          this.scopedClusterClient.asSecondaryAuthUser.ingest.putPipeline(
-            getSLOSummaryPipelineTemplate(slo, this.spaceId, this.basePath)
-          ),
-        { logger: this.logger }
+      const summaryPipelinePromise = this.createPipeline(
+        getSLOSummaryPipelineTemplate(slo, this.spaceId, this.basePath)
       );
+
       rollbackOperations.push(() =>
-        this.scopedClusterClient.asSecondaryAuthUser.ingest.deletePipeline(
-          { id: getSLOSummaryPipelineId(slo.id, slo.revision) },
-          { ignore: [404] }
-        )
+        this.deletePipeline(getSLOSummaryPipelineId(slo.id, slo.revision))
       );
 
-      await this.summaryTransformManager.install(slo);
+      const summaryTransformPromise = this.summaryTransformManager.install(slo, true);
       rollbackOperations.push(() => this.summaryTransformManager.uninstall(summaryTransformId));
-
-      await this.summaryTransformManager.start(summaryTransformId);
       rollbackOperations.push(() => this.summaryTransformManager.stop(summaryTransformId));
 
-      await retryTransientEsErrors(
-        () =>
-          this.esClient.index({
-            index: SLO_SUMMARY_TEMP_INDEX_NAME,
-            id: `slo-${slo.id}`,
-            document: createTempSummaryDocument(slo, this.spaceId, this.basePath),
-            refresh: true,
-          }),
-        { logger: this.logger }
-      );
+      const tempDocPromise = this.createTempSummaryDocument(slo);
+
+      await Promise.all([
+        sloPipelinePromise,
+        rollupTransformPromise,
+        summaryPipelinePromise,
+        summaryTransformPromise,
+        tempDocPromise,
```

Comment:
We need to add a rollback operation for the temp document.
Previously it was not handled since it was the last operation to happen, and if it was erroring, nothing was required to be done in order to revert it. 
But now it can happen first, and another operation could fail. Resulting in the temp doc being left over.